sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -108,49 +114,55 @@ internal static IEnumerable<BatchedLogs<T>> Batch<T>(IEnumerable<T> logEntries,
                 var memory = entry.ToMemory();
                 if (memory.Length > SingleUploadThreshold) // if single log is > 1 Mb send to be gzipped by itself
                 {
-                    MemoryStream tempStream = new MemoryStream(); // create tempStream for individual log
-                    WriteMemory(tempStream, BinaryData.FromString("[").ToMemory());
-                    WriteMemory(tempStream, memory);
-                    WriteMemory(tempStream, BinaryData.FromString("]").ToMemory());
-                    tempStream.Position = 0;
-                    yield return new BatchedLogs<T>(new List<T>{log}, BinaryData.FromStream(tempStream));
+                    // Create tempWriter and tempStream for individual log
+                    MemoryStream tempStream = new MemoryStream();
+                    Utf8JsonWriter tempWriter = new Utf8JsonWriter(tempStream);
```

Comment:
Would it be possible to just us a single Utf8JsonWriter and flush and reset as needed?